src/django/api/views/facility/facility_claim_view_set.py

```diff
@@ -417,6 +418,42 @@ def get_claimed_details(self, request, pk=None):
                 else:
                     setattr(claim, field_name, None)
 
+            for date_field in ('opening_date', 'closing_date'):
+                value = request.data.get(date_field)
+                if value in (None, ''):
+                    setattr(claim, date_field, None)
+                else:
+                    try:
+                        setattr(
+                            claim,
+                            date_field,
+                            datetime.fromisoformat(value).date(),
+                        )
+                    except (ValueError, TypeError):
+                        setattr(claim, date_field, None)
+
+            emission_fields = (
+                'estimated_annual_throughput',
+                'energy_coal',
+                'energy_natural_gas',
+                'energy_diesel',
+                'energy_kerosene',
+                'energy_biomass',
+                'energy_charcoal',
+                'energy_animal_waste',
+                'energy_electricity',
+                'energy_other',
+            )
+            for field_name in emission_fields:
+                value = request.data.get(field_name)
+                if value in (None, ''):
+                    setattr(claim, field_name, None)
+                else:
+                    try:
+                        setattr(claim, field_name, int(value))
+                    except (ValueError, TypeError):
+                        setattr(claim, field_name, None)
```

Comment:
I would advice us to avoid nested statement as much as possible, this could potentially be re-written to something like:

```python
for field_name in emission_fields:
    value = request.data.get(field_name, None)

    if not value:
        setattr(claim, field_name, None)
        continue

    try:
        setattr(claim, field_name, int(value))
    except (ValueError, TypeError):
        setattr(claim, field_name, None)
```

Reply:
Good suggestion, I refactored this part of code.